### DISCUSSION TITLE
Using .env.local file. Getting process.env.NEXT_PUBLIC_VERCEL_ENV is undefined on client

### DISCUSSION THREAD

**Discussion by cbdeveloper** [vercel/vercel#6311] [Help] [answered]
I have a `.env.local` file that I've create with Vercel CLI. But my client code in my local server is reading `process.env.NEXT_PUBLIC_VERCEL_ENV` as `undefined`. What could be the reason of that? The server code reads the `process.env.VERCEL_ENV` just fine. I'm starting it with `npm run dev`, which runs `next dev -p 80`.

**Comment by cbdeveloper** [comment, marked as answer]
From: https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser

![image](https://user-images.githubusercontent.com/43407798/120467138-c35c0d80-c397-11eb-9e01-6d84e3e42c24.png)

So you have to add `NEXT_PUBLIC_VERCEL_ENV` to `.env.local` in order to read it from the client. This was somehow unexpected, because I though the `NEXT_PUBLIC_VERCEL_ENV` would be automatically added just like on regular deployments.

I mean the behavior from: https://vercel.com/docs/environment-variables#system-environment-variables

![image](https://user-images.githubusercontent.com/43407798/120467505-2b125880-c398-11eb-812b-f2d122816c54.png)

![image](https://user-images.githubusercontent.com/43407798/120467475-2057c380-c398-11eb-87df-f5af57f2777f.png)

**Comment by pk058** [comment]
@cbdeveloper  
I am facing the same issue. In my case, I manually created a .env.local file with NEXT_PUBLIC_VERCEL_ENV_API_KEY. I have added my Preview and Production API Key (They are different for both modes) with the Same Name on Vercel Dashboard. It is building successfully on local but getting an error (API KEY is Invalid) while building on Vercel.
Do I need to have a .env.development and .env.production file in my project or just adding keys on Vercel Dashboard is enough?
